prov/usnic/src/usdf_fabric.c

```diff
@@ -75,12 +75,82 @@
 
 struct usdf_usnic_info *__usdf_devinfo;
 
+char *
+get_devname(uint32_t version, struct usd_device_attrs *dap)
+{
+	char *bufp;
+	struct in_addr in;
+	char *addrnetw;
+	size_t size;
+
+	if (FI_VERSION_GE(version, FI_VERSION(1, 4))) {
+		in.s_addr = dap->uda_ipaddr_be & dap->uda_netmask_be;
+		addrnetw = inet_ntoa(in);
+		size = snprintf(NULL, 0, "%s/%d", addrnetw, dap->uda_prefixlen) + 1;
+		bufp = calloc(1, size);
+		sprintf(bufp, "%s/%d", addrnetw, dap->uda_prefixlen);
+	} else {
+		bufp = strdup(dap->uda_devname);
+	}
+
+	return bufp;
+}
+
+enum check_devname_func_t {
+	CHECK_DEVNAME_GETINFO,
+	CHECK_DEVNAME_FABRIC
+};
+
+int
+check_devname(uint32_t version, struct usd_device_attrs *dap, struct fi_fabric_attr *fattrp,
+		enum check_devname_func_t caller)
+{
+	int ret;
+	char *devname;
+
+	char* devname_1_3 = get_devname(FI_VERSION(1,3), dap);
+	char* devname_1_4 = get_devname(FI_VERSION(1,4), dap);
```

Comment:
You should only call these `get_devname()` for each version when/if you need to.  I.e., put them inside the version checks, below.